chore: adds data structures for container dependency

xar-tol opened the pull request:
Incorporates data structures for container dependencies to address #2175.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

xar-tol committed:
chore: adds data structures for container dependency

Incorporates data structures for container dependencies to address #2175.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bvtujo commented on templates/workloads/partials/cf/sidecars.yml:
```diff
@@ -49,6 +49,10 @@
   DockerLabels:{{range $name, $value := $sidecar.DockerLabels}}
     {{$name | printf "%q"}}: {{$value | printf "%q"}}{{end}}
 {{- end -}}
+{{- if $sidecar.Dependencies}}
```
It looks like the CloudFormation template for container dependencies documented [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-dependson) uses the `DependsOn` key and requires a list of [`ContainerDependency` objects](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdependency.html), rather than a map. Can we update this and the WorkloadContainer CF templates to match the cloudformation spec?

xar-tol replied:
Adjusted to better match the DependsOn / Condition / ContainerName format.

bvtujo reviewed (commented):
Yay!!! This looks really good--great unit tests in the `manifest` package! 

One comment, however, about the Cloudformation spec for container dependencies.

xar-tol committed:
chore: adjusts cf templates to match documentation

Adjusts cf templates to better match documentation format.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bvtujo reviewed (approved):
Looks great to me!

huanjani commented on internal/pkg/manifest/workload_test.go:
```diff
@@ -579,3 +579,60 @@ network:
+		wantedStruct Image
+		wantedError  error
+	}{
+		"Unspecified optional dependencies don't appear im image": {
```
tiny nit:
```suggestion
		"Unspecified optional dependencies don't appear in image": {
```

xar-tol replied:
Whoops! Fixed.

xar-tol committed:
chore: fixes spelling error

im->in

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

huanjani commented on templates/workloads/partials/cf/workload-container.yml:
```diff
@@ -11,6 +11,13 @@
+{{- if .DependsOn}}
+  DependsOn:
+  {{- range $name, $conditionFrom := .DependsOn}}
+  - Condition: {{$conditionFrom}}
+    ContainerName: {{$name}}
```
These yaml templates are super tricky and annoying because they will compile without complaining about formatting, but will fail later. 😤 
These two lines need to be indented by two spaces.

xar-tol replied:
Added two extra spaces for each  to this and the sidecars.yml file.

bvtujo commented on templates/workloads/partials/cf/sidecars.yml:
```diff
@@ -49,6 +49,13 @@
+{{- if $sidecar.DependsOn}}
+  DependsOn:
+  {{- range $name, $conditionFrom := $sidecar.DependsOn}}
+  - Condition: {{$conditionFrom}}
```
Same here to Janice's comment 😤 nice catch, @huanjani

xar-tol replied:
Adjusted with 2 spaces!

bvtujo commented on internal/pkg/manifest/workload_test.go:
```diff
@@ -579,3 +579,60 @@ network:
+				require.EqualError(t, err, tc.wantedError.Error())
+			} else {
+				require.NoError(t, err)
+				// check memberwise dereferenced pointer equality
```
I don't think either of these are pointers; we can probably remove this comment.

xar-tol replied:
Removed.

xar-tol committed:
chore: adds spaces, removes comment

adds extra indentation to Condition/ContainerName for DependsOn in .yml files
removes comment from workload_test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

huanjani reviewed (approved):
Looks fantastic! Nice work!

mergify[bot] committed:
Merge branch 'mainline' into xar-to/adds-sidecar-dependecies-to-manifest
